### PR TITLE
deprecate /servers/checks and move functionality to /servercheck

### PR DESCRIPTION
**Pull request by mattjackson220**
## What does this PR (Pull Request) do?

- [x] This PR is not related to any Issue 

This PR deprecates /servers/checks and moves functionality to /servercheck

## Which Traffic Control components are affected by this PR?

- Documentation
- Traffic Ops
- Traffic Portal

## What is the best way to verify this PR?

Verify that GET /api/1.5/servers/checks now has an appropriate deprecation alert
Verify that GET /api/2.0/servers/checks does not exist
Verify that GET /api/2.0/servercheck returns the same information that /api/1.5/servers/checks does but without the alert
Verify that Traffic Portal Cache Checks link still works (requires TP to use api 2.0)

## The following criteria are ALL met by this PR

- [x] This PR does not need testing updates
- [x] This PR includes documentation
- [x] This PR includes an update to CHANGELOG.md
- [x] This PR includes any and all required license headers
- [x] This PR does not include a database migration
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

### CODE REVIEW

File: traffic_ops/traffic_ops_golang/servercheck/servercheck.go [hunk 1]
```diff
@@ -205,57 +208,79 @@ func ReadServersChecks(w http.ResponseWriter, r *http.Request) {
...
+
+	api.WriteResp(w, r, data)
+}
+
+// DeprecatedReadServersChecks is the handler for deprecated GET requests for /servers/checks
+func DeprecatedReadServersChecks(w http.ResponseWriter, r *http.Request) {
+	inf, userErr, sysErr, errCode := api.NewInfo(r, nil, nil)
+	tx := inf.Tx.Tx
+	if userErr != nil || sysErr != nil {
+		api.HandleDeprecatedErr(w, r, tx, errCode, userErr, sysErr, util.StrPtr(ServerCheck_Get_Endpoint))
```
Comment (ocket8888) [inline traffic_ops/traffic_ops_golang/servercheck/servercheck.go]: instead of calling `util.StrPtr` couldn't you just give it `&ServerCheck_Get_Endpoint`?

**Comment by ocket8888** [comment]
Also this endpoint is used in Traffic Portal at `traffic_portal/app/src/common/api/ServerService.js:212:45` - can you update that to the new endpoint name?

File: traffic_ops/traffic_ops_golang/servercheck/servercheck.go (hunk 1 above)
Comment (mattjackson220) [reply traffic_ops/traffic_ops_golang/servercheck/servercheck.go]: Go doesnt let you do & to a const.  so i went this route to keep it a const, but im also open to changing it to a var and then doing &.

**Comment by mattjackson220** [comment]
> Also this endpoint is used in Traffic Portal at `traffic_portal/app/src/common/api/ServerService.js:212:45` - can you update that to the new endpoint name?

I'm not sure i follow. line 212 is caches/stats which is different right? the update at line 223 should have been enough?

File: traffic_ops/traffic_ops_golang/servercheck/servercheck.go (hunk 1 above)
Comment (ocket8888) [reply traffic_ops/traffic_ops_golang/servercheck/servercheck.go]: oh, right. I always forget about that, it's so weird.

**Comment by ocket8888** [comment]
Yeah, that'll do it.

File: traffic_ops/traffic_ops_golang/servercheck/servercheck.go [hunk 2]
```diff
@@ -205,57 +208,79 @@ func ReadServersChecks(w http.ResponseWriter, r *http.Request) {
 	}
 	defer inf.Close()
 
+	data, userErr, sysErr, errCode := handleReadServerCheck(inf, tx)
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, tx, errCode, userErr, sysErr)
```
Comment (mhoppa) [inline traffic_ops/traffic_ops_golang/servercheck/servercheck.go]: need a return here

File: traffic_ops/traffic_ops_golang/servercheck/servercheck.go [hunk 3]
```diff
@@ -205,57 +208,79 @@ func ReadServersChecks(w http.ResponseWriter, r *http.Request) {
...
+	tx := inf.Tx.Tx
+	if userErr != nil || sysErr != nil {
+		api.HandleDeprecatedErr(w, r, tx, errCode, userErr, sysErr, util.StrPtr(ServerCheck_Get_Endpoint))
+		return
+	}
+	defer inf.Close()
+
+	data, userErr, sysErr, errCode := handleReadServerCheck(inf, tx)
+	if userErr != nil || sysErr != nil {
+		api.HandleDeprecatedErr(w, r, tx, errCode, userErr, sysErr, util.StrPtr(ServerCheck_Get_Endpoint))
```
Comment (mhoppa) [inline traffic_ops/traffic_ops_golang/servercheck/servercheck.go]: and here

File: traffic_ops/traffic_ops_golang/servercheck/servercheck.go (hunk 2 above)
Comment (mattjackson220) [reply traffic_ops/traffic_ops_golang/servercheck/servercheck.go]: done

File: traffic_ops/traffic_ops_golang/servercheck/servercheck.go (hunk 3 above)
Comment (mattjackson220) [reply traffic_ops/traffic_ops_golang/servercheck/servercheck.go]: done

File: docs/source/api/v1/servers_checks.rst [hunk 4]
```diff
@@ -61,7 +64,13 @@ Response Structure
 	Date: Thu, 23 Jan 2020 19:00:19 GMT
 	Content-Length: 202
```
Comment (shamrickus) [inline docs/source/api/v1/servers_checks.rst]: Incorrect content length

File: docs/source/api/v2/servercheck.rst [hunk 5]
```diff
@@ -19,11 +19,73 @@
...
+	Access-Control-Allow-Credentials: true
+	Access-Control-Allow-Headers: Origin, X-Requested-With, Content-Type, Accept, Set-Cookie, Cookie
+	Access-Control-Allow-Methods: POST,GET,OPTIONS,PUT,DELETE
+	Access-Control-Allow-Origin: *
+	Content-Encoding: gzip
+	Content-Type: application/json
+	Set-Cookie: mojolicious=...; Path=/; Expires=Thu, 23 Jan 2020 20:00:19 GMT; Max-Age=3600; HttpOnly
+	X-Server-Name: traffic_ops_golang/
+	Date: Thu, 23 Jan 2020 19:00:19 GMT
+	Content-Length: 202
```
Comment (shamrickus) [inline docs/source/api/v2/servercheck.rst]: Wrong Content-Length

File: docs/source/api/v1/servers_checks.rst (hunk 4 above)
Comment (mattjackson220) [reply docs/source/api/v1/servers_checks.rst]: done

File: docs/source/api/v2/servercheck.rst (hunk 5 above)
Comment (mattjackson220) [reply docs/source/api/v2/servercheck.rst]: done

**Pull request merged**